src/pages/index.js

```diff
@@ -54,43 +58,246 @@ const addFormValidator = new FormValidator(
+function handleDeleteClick(cardId) {
+  if (!cardId || typeof cardId !== "string") {
+    console.error("Invalid cardId for deletion:", cardId);
+    return;
+  }
+  deleteCardPopup.open({ cardId: cardId });
+}
 
 function handleImageClick(data) {
   imagePopup.open(data);
 }
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const sanitizedLink =
+    cardData.link.replace(/["'\s]+$/, "").trim() ||
+    "https://placehold.co/150x150";
+  const card = new Card(
+    {
+      name: cardData.name,
+      link: sanitizedLink,
+      _id: cardData._id || null,
+    },
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+
   const element = card.getView();
+  if (!cardData._id) {
+    console.warn("Card created with no _id, deletion might fail:", cardData);
+  } else {
+    const initialIsLiked =
+      cardData.isLiked !== undefined ? cardData.isLiked : false;
+    likeStatuses.set(cardData._id, initialIsLiked);
+    const likeButton = element.querySelector(".card__like-button");
+    if (likeButton) {
+      likeButton.classList.toggle("card__like-button_active", initialIsLiked);
+    }
+  }
   return element;
 }
 
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo({
-    name: data["edit-profile"],
-    job: data["profile-description"],
-  });
-  profileEditPopup.close();
+  const saveButton = profileEditForm.querySelector(".modal__save-button");
+  saveButton.textContent = "Saving...";
+  saveButton.classList.add("loading");
+  saveButton.disabled = true;
+  api
+    .updateUserInfo(data["edit-profile"], data["profile-description"])
+    .then((updatedInfo) => {
+      userInfo.setUserInfo({
+        name: updatedInfo.name,
+        about: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+      saveButton.classList.remove("loading");
+      saveButton.disabled = false;
+    });
 }
 
 function handleAddCardFormSubmit(data) {
-  const newCardElement = createCard({ name: data.title, link: data.url });
-  section.addItem(newCardElement);
-  addCardFormElement.reset();
-  addFormValidator.disableButton();
-  addCardPopup.close();
+  const sanitizedUrl = data.url.replace(/["'\s]+$/, "").trim();
+  const saveButton = addCardFormElement.querySelector(".modal__save-button");
+  saveButton.textContent = "Saving...";
+  saveButton.classList.add("loading");
+  saveButton.disabled = true;
+  api
+    .addCard(data.title, sanitizedUrl)
+    .then((newCardInfo) => {
+      const newCardElement = createCard({
+        name: newCardInfo.name,
+        link: newCardInfo.link,
+        _id: newCardInfo._id,
+      });
+      section.addItem(newCardElement);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submission. It’s a **waste of resources**. You need to call `toggleButtonState` there.

Reply:
`addFormValidator.toggleButtonState();` should be used after submission instead of `resetValidation `